test/api/AssetTest.ts

```diff
@@ -239,6 +249,107 @@ describe('Asset', function() {
+          testData.newAsset
+        );
+      });
+
+      it('Should be able to create an ioThink asset connector', async () => {
+        const settings = await testData.centralUserService.settingApi.readAll({});
+        let newSetting = false;
+        let setting: SettingDB = settings.data.result.find((s) => s.identifier === TenantComponents.ASSET);
+        if (!setting) {
+          setting = {} as SettingDB;
+          setting.identifier = TenantComponents.ASSET;
+          newSetting = true;
+        }
+        const settingID = '0123456789abcdef';
+        setting.content = {
+          type: AssetSettingsType.ASSET,
+          asset: {
+            connections: [{
+              id: settingID,
+              name: 'Testing IoThink Asset Connector',
+              description: 'Testing IoThink Asset Connector',
+              url: config.get('assetConnectors.ioThink.url'),
+              timestamp: (new Date()),
+              type: AssetConnectionType.IOTHINK,
+              refreshIntervalMins: 1,
+              iothinkConnection: {
+                user: config.get('assetConnectors.ioThink.user'),
+                password: config.get('assetConnectors.ioThink.password')
+              }
+            }]
+          }
+        };
+        setting.sensitiveData = ['content.asset.connections[0].iothinkConnection.password'];
+        if (newSetting) {
+          await testData.centralUserService.settingApi.create(setting);
+        }
+        await testData.centralUserService.settingApi.update(setting);
+        const settingsUpdated = await testData.centralUserService.settingApi.readAll({});
+        const settingUpdated: SettingDB = settingsUpdated.data.result.find((s) => s.identifier === TenantComponents.ASSET);
+        expect(settingUpdated.content.asset.connections[0].id).to.be.eq(settingID);
+        testData.ioThinkAssetConnectorID = settingID;
+      });
+
+      it('should be able to connect ioThink asset connector connection', async () => {
+        const response = await testData.centralUserService.assetApi.checkAssetConnectorLink(testData.ioThinkAssetConnectorID);
+        expect(response.data?.status).to.be.eq('Success');
+        expect(response.data?.connectionIsValid).to.be.eq(true);
+      });
+
+      it('should be able to connect existing asset to asset connector', async () => {
+        // Change entity
+        testData.newAsset.dynamicAsset = true;
+        testData.newAsset.meterID = config.get('assetConnectors.ioThink.meterID');
+        testData.newAsset.connectionID = testData.ioThinkAssetConnectorID;
+        // Update
+        await testData.userService.updateEntity(
+          testData.userService.assetApi,
+          testData.newAsset
+        );
+      });
+
+      it('should be able to retrieve latest consumption', async () => {
+        const response = await testData.centralUserService.assetApi.retrieveLatestConsumption(testData.newAsset.id);
+        expect(response.data?.status).to.be.eq('Success');
```

Comment:
Please also check the response properties returned

Reply:
There are no other properties except status in the data field here. This checks if the latest consumption is retrieved from ioThink and stored. Post storage, a success response is sent.

Reply:
Okay, should be fine then